### PR TITLE
Map special rebalance and several location ports

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary

SUMMARY: Balance "More interesting and flavorful maps"

#### Purpose of change

Port https://github.com/CleverRaven/Cataclysm-DDA/pull/46701 to make BN maps more interesting and fun

#### Describe the solution

So far just a straight port, removing wallet content. 
Note: Currently new lab maps aren't spawning on science bodies, good to figure that out without wallets
Note 2: Looks like this is scooping up a bunch of map special changes from DDA that may or not be good to port as part of this. 
Those are:
    "id": "Pottery Cottage",  [all new, SKIPPED ]
    "id": "Farm_2side",  [some new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/45059 ]
    "id": "Farm_stills", [some new, SKIPPED ]
    "id": "St_Johns_farm", [some new, ported here, removing zombie cows]
    "id": "FEMA_Camp2", [new ID, SKIPPED]
    "id": "Prison", [some new], ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/41372/files 
    "id": "Prison Hidden Lab", [some new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/41372/files ]
    "id": "ranch_camp", [some new, SKIPPED ]
    "id": "Hot Springs", [all new, SKIPPED]
    "id": "Mansion_Road", [some new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/44250 ]
    "id": "Mansion_Road_Alt", [some new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/44250 ]
    "id": "Mansion_Wild", [some new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/44250 ]
    "id": "Mansion_WildAlt", [some new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/44250 ]
    "id": "double_microlab", [some new, SKIPPED]
    "id": "o_airport", [all new, ported here copying https://github.com/CleverRaven/Cataclysm-DDA/pull/40901 ]
    "id": "o_lightindustry", [all new, SKIPPED needs https://github.com/CleverRaven/Cataclysm-DDA/pull/41731 or patching]
    "id": "Freshwater Research Station", [all new, SKIPPED]
    "id": "Bastion Fort", [all new, SKIPPED]
Grabbed the vehicle group changes from https://github.com/CleverRaven/Cataclysm-DDA/pull/41450 to troubleshoot an airport error
Skipped central lab changes that broke BN ENDGAME code

#### Describe alternatives you've considered

N/A

#### Testing

IT'S ALIVE. Game starts no errors, mapgen looks good.

#### Additional context

I've done a lot of playtesting with this in DDA and think it's a lot of fun and matches the gameplay goals of BN nicely. Kudos to Venera3 for the good work

### CODE REVIEW

**Comment by LyleSY** [comment]
Travis is looking relatively good right now. Still lots to check left, but what's here looks pretty stable and good

**Comment by LyleSY** [comment]
The failed test is unrelated. This ended up way bigger than I planned. Is it possible to get a clean test?

**Comment by Coolthulhu** [comment]
Needs fixing.
You should test a PR on your own side before removing the [WIP]. At least check if it loads - the current version doesn't load.

**Comment by olanti-p** [comment]
It may be better to split off mapgen changes into separate PR(s).
There are a lot of them, and some use DDA-exclusive content like ramps (which implies z-level vehicle transition may be factored into location design, and wouldn't work in BN).
<details>
<summary> Partial list of errors, after fixing GROUP_ZOMBIE_BOVINE reported by Transifex </summary>

```
13:08:25.940 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid terrain id "t_ramp_down_low"
13:08:25.947 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid terrain id "t_ramp_down_low"
13:08:25.958 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid terrain id "t_ramp_down_low"
13:08:25.967 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid terrain id "t_ramp_down_low"
13:08:25.976 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid terrain id "t_ramp_up_high"
13:08:25.983 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid terrain id "t_ramp_down_low"
13:08:26.253 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_electronic"
13:08:26.258 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_light"
13:08:26.263 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_heavy"
13:08:26.268 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_old"
13:08:26.275 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_electronic"
13:08:26.281 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_light"
13:08:26.285 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_heavy"
13:08:26.290 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_old"
13:08:26.296 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_electronic"
13:08:26.302 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_light"
13:08:26.306 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_heavy"
13:08:26.312 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_old"
13:08:26.319 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_electronic"
13:08:26.323 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_light"
13:08:26.329 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_heavy"
13:08:26.333 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_old"
13:08:26.340 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_electronic"
13:08:26.346 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_light"
13:08:26.351 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_heavy"
13:08:26.356 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_old"
13:08:26.362 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_electronic"
13:08:26.367 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_light"
13:08:26.372 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_heavy"
13:08:26.377 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid furniture id "f_machinery_old"
13:08:32.508 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_2nd".
13:08:32.516 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_2nd".
13:08:32.522 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_2nd".
13:08:32.528 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_2nd".
13:08:32.535 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_roof".
13:08:32.541 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_roof".
13:08:32.547 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_roof".
13:08:32.552 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field_roof".
13:08:32.557 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field".
13:08:32.561 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field".
13:08:32.566 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field".
13:08:32.571 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:828 [check_consistency] No mapgen terrain exists for "pottery_cottage_field".
13:08:32.583 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid overmap terrain id "s_lot_north"
13:08:32.587 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid overmap terrain id "s_lot_north"
13:08:32.592 ERROR : D:\git\cataclysm-bn-b\src\overmap.cpp:1014 [check] In overmap special "Mine Entrance", terrain "s_lot_north" is invalid.
13:08:32.936 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:457 [convert] invalid overmap terrain id "s_lot_north"
13:08:32.941 ERROR : D:\git\cataclysm-bn-b\src\generic_factory.h:429 [obj] invalid overmap terrain id "s_lot_north"
```

</details>

Considering the rebalance PR changes are mostly to a single file and are concentrated on oms spawn chances/conditions/flags, you could instead of adding new locations bump spawn chances of existing similar ones.

**Comment by LyleSY** [comment]
OH! The mod error made me think the group zombie bovine thing was unrelated. Not so, just hidden in a file I hadn't combed through. I've set up a github instance for BN so I can properly test things now

**Comment by LyleSY** [comment]
I don't understand enough about that overmap test that is failing. Is that related?

**Comment by Coolthulhu** [comment]
> I don't understand enough about that overmap test that is failing. Is that related?

Probably not. I restarted the test.
If the problem was caused by your PR, it probably would cause json loading errors.

**Comment by Coolthulhu** [comment]
Wow, this time that mapgen error actually is related and caused here!
Looks like you accidentally removed an `"ENDGAME"` flag from central lab.

**Comment by Coolthulhu** [comment]
But it's not as simple as adding it back, because of hardcode. Looks like chances for central lab need to stay on 1.

File: data/json/items/book/maps.json [hunk 1]
```diff
@@ -10,10 +10,33 @@
...
-    "material": "paper",
+    "material": [ "paper" ],
     "symbol": ";",
     "color": "light_gray"
   },
+  {
+    "id": "labmap",
+    "copy-from": "abstractmap",
+    "type": "GENERIC",
+    "name": { "str": "handwritten note" },
```
Comment (Coolthulhu) [inline data/json/items/book/maps.json:21]: Needs a rename. No "inconspicuously named items" in BN. It doesn't need to explicitly mention labs, but it should at least have "map" in name.

File: data/json/mapgen/farm.json [hunk 2]
```diff
@@ -93,6 +93,7 @@
         { "monster": "GROUP_ZOMBIE", "x": [ 4, 4 ], "y": [ 14, 14 ], "repeat": [ 1, 6 ], "density": 0.2 },
         { "monster": "GROUP_DOMESTIC", "x": [ 10, 12 ], "y": [ 15, 17 ], "repeat": [ 1, 2 ], "density": 0.2 }
       ],
+      "sealed_item": { "D": { "item": { "item": "seed_corn", "chance": 70 }, "furniture": "f_plant_seedling" } },
```
Comment (Coolthulhu) [inline data/json/mapgen/farm.json:96]: This doesn't look right, it looks like it will create empty seedlings at 30% chance.

Comment (Coolthulhu) [reply data/json/mapgen/farm.json:96]: DDA had an update to its `sealed_item` function. I'll have to port that.
In the meanwhile, you should move the `"chance"` from `"item"` to the outside object, in this case `"D"`. And do the same for other `"sealed_item"`s. After I port the DDA version, those would throw errors anyway.

File: data/json/mapgen/farm.json [hunk 3]
```diff
@@ -300,6 +303,17 @@
...
+      "sealed_item": {
+        "D": { "item": { "item": "seed_corn", "chance": 70 }, "furniture": "f_plant_seedling" },
+        "♥": { "item": { "item": "seed_tomato", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "♠": { "item": { "item": "seed_lettuce", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "♦": { "item": { "item": "seed_cucumber", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "♣": { "item": { "item": "seed_carrot", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "Ʌ": { "item": { "item": "seed_zucchini", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "Δ": { "item": { "item": "seed_celery", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "‡": { "item": { "item": "seed_pumpkin", "chance": 100 }, "furniture": "f_planter_harvest" },
+        "%": { "item": { "item": "seed_oats", "chance": 100 }, "furniture": "f_planter_harvest" }
```
Comment (Coolthulhu) [inline data/json/mapgen/farm.json:315]: `"chance": 100` should be redundant - they're 100 by default.

**Comment by Coolthulhu** [comment]
Alright, now the seedling chance thing is up. You'll now get proper load error messages in the game if you try to load it with your PR.

**Comment by LyleSY** [comment]
CLANG calendar build error does not look related

**Comment by Coolthulhu** [comment]
After multiple fixes, it's in. Good job at surviving my perfectionist nitpicking.

**Comment by LyleSY** [comment]
Big learning project, thanks for sticking with me

**Pull request merged**